src/Symfony/Component/Console/Command/Command.php

```diff
@@ -83,14 +83,35 @@ public static function getDefaultDescription(): ?string
     }
 
     /**
-     * @param string|null $name The name of the command; passing null means it must be set in configure()
+     * @param string|callable|null $name The name of the command or an invokable object that uses AsCommand attribute; passing null means it must be set in configure()
      *
      * @throws LogicException When the command name is empty
      */
-    public function __construct(?string $name = null)
+    public function __construct(null|callable|string $name = null)
```

Comment:
We don't have to keep the parameter name, they are not covered by out BC promise (except for attribute constructors).

@yceruto suggested `$subject` https://github.com/symfony/symfony/pull/60823#discussion_r2154807468.

On a 2nd thought, we can have a 2nd parameter named `$code` that can be any callable, and would require to be an invokable object with the `#[AsCommand]` attribute only when `$name` is `null`.

```suggestion
    public function __construct(?string $name = null, ?callable $code = null)
```

Reply:
I like the 2nd approach, added a commit that introduces it.